lib/recurly.d.ts

```diff
@@ -8,2381 +8,2381 @@ export interface Site {
   /**
    * Site ID
    */
-  id: string | null;
+  id?: string | null;
```

Comment:
I had to make all these optional so that they could be constructed properly:

![Screen Shot 2020-07-14 at 12 53 15 PM](https://user-images.githubusercontent.com/185919/87459956-b55f9b80-c5d1-11ea-9674-752a792ba611.png)

@dbrudner does this seem correct?

Reply:
Yeah definitely, this is a great QOL improvement in general, and especially for TS apps with [strictNullChecks] (https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#--strictnullchecks) enabled.

The type lgtm. This lets users pass in either a string, explicit `null`, or leave off the parameter completely. Assuming the runtime allows for those three possible values, seems solid.